instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ProcessResource.java

```diff
@@ -69,10 +72,33 @@ private static Resource doBuildResource() {
 
       attributes.put(ResourceAttributes.PROCESS_EXECUTABLE_PATH, executablePath.toString());
 
+      List<String> commandArgs = new ArrayList<>();
```

Comment:
Consider using https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ProcessHandle.Info.html With that you should be able to get what is needed on jdk9. If we have a good implementation for jdk9+ then perhaps it isn't too important that our jdk8 implementation isn't as good.

Reply:
Alternatively, other resource providers use MR jar: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/resources/library/src/main/java9/io/opentelemetry/instrumentation/resources/ProcessArguments.java

Reply:
I've added @mateuszrzeszutek suggestion when it's available. Still falls back to the current implementation though.